### PR TITLE
PS-6849 Add JUnit reporting support to MTR (5.6)

### PR DESCRIPTION
**Pull request by oleksandr-kachan**
- Backport MTR shutdown report implementation from 5.7 - github.com/mysql/mysql-server/commit/8aeda369bf1e597fafee9dcd5b66aa148a8b79eb
- Add JUnit reporting support to the MySQL test suite runner
- Add "junit" test suite with deliberately failing tests
- Add a deliberately failing unittests suite
- Add unittests timeout, include CTest output into JUnit report
- Attach mysqld shutdown and valgrind reports to JUnit report
- Run unit tests under Valgrind too if it was enabled
- Mark self-skipping tests as skipped in JUnit reports
- Convey tests output to JUnit reports regardless of test failure
- Exclude tests skipped by framework from results xml

### CODE REVIEW

File: mysql-test/mysql-test-run.pl [hunk 1]
```diff
@@ -6649,7 +6680,7 @@ sub valgrind_arguments {
 #
 
 sub valgrind_exit_reports() {
```
Comment (percona-ysorokin) [inline mysql-test/mysql-test-run.pl:6695]: After the change in this function MTR prints a a big number of
```
Argument "" isn't numeric in exit at /home/yura/ws/percona-server/mysql-test/mysql-test-run.pl line 1055.
```
errors when run with the `--valgrind` option

**Comment by oleksandr-kachan** [comment]
https://ps56.cd.percona.com/job/percona-server-5.6-param/128/

**Comment by percona-ysorokin** [comment]
@oleksandr-kachan 
Regular Debug
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1336/
and ASan
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1337/
configurations look good
However, Valgrind build does not succeed
There is no MTR summary at the end (including the list of failing test cases `Failing test(s): ...`) and probably wrong error code is returned.
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1338/consoleText

**Comment by percona-ysorokin** [comment]
@oleksandr-kachan 
In the `Standard output` section of some tests there are multiple `CURRENT_TEST:` lines.
Some of them refer to invalid test name.
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1336/testReport/junit/centos_7.Debug/junit/06_bootstrap_crash/
and
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1336/testReport/junit/centos_7.Debug/junit/10_immediate_hang/

**Comment by oleksandr-kachan** [comment]
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1339/
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1340/
https://ps3.cd.percona.com/view/5.6/job/percona-server-5.6-pipeline/1341/
https://ps56.cd.percona.com/job/percona-server-5.6-pipeline/9220/

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**